changelog/unreleased/issue-2298

```diff
@@ -0,0 +1,10 @@
+Bugfix: Do not hang when run as a background job
+
+Restic did hang on exit when it was started as a background job, for example
+using `restic ... &`. Restic was stopped when trying to restore the terminal
+configuration on exit to ensure that reading a password from the terminal
+cannot leave the user with a misconfigured terminal. We have changed the
+behavior to only restore the terminal configuration when restic is interrupted
+while reading a password from the terminal.
```

Comment:
With the risk of being picky, I think this is too technical for the average user. If they want to deep dive they can check the discussions here on GitHub. What do you think about:

Restic used to hang on exit when it was started as a background job, for example
using `restic ... &`. This has been fixed by now only trying to restore the terminal
when restic is being interrupted while reading a password from the terminal.

Reply:
My changelog text felt too long while writing it, I think its in fact better to leave the technical details. I've changed it to your proposal and added a remark that the hang was caused by the terminal reconfiguration. Otherwise its not really clear how the two sentences relate to each other.

Reply:
I'm not seeing the remark, where did you add that?

Reply:
I've sneaked it into the first sentence: `hang on exit while restoring the terminal configuration when`.